FreeBSD doesn't ship libatomic, so use __sync* builtins instead:

media_driver/CMakeFiles/iHD_drv_video_OBJ.dir/linux/common/os/mos_context_specific.cpp.o: In function `OsContextSpecific::SetSliceCount(unsigned int*)':
media_driver/linux/common/os/mos_context_specific.cpp:441: undefined reference to `__atomic_load_8'
media_driver/linux/common/os/mos_context_specific.cpp:444: undefined reference to `__atomic_store_8'
c++: error: linker command failed with exit code 1 (use -v to see invocation)

https://bugs.freebsd.org/bugzilla/show_bug.cgi?id=220822

--- media_driver/linux/common/os/mos_context_specific.cpp.orig	2018-10-01 21:39:52 UTC
+++ media_driver/linux/common/os/mos_context_specific.cpp
@@ -437,6 +437,10 @@ void OsContextSpecific::SetSliceCount(uint32_t *pSlice
 
         for (int sliceCountShm = m_gtSystemInfo.SliceCount; sliceCountShm > 0; sliceCountShm--)
         {
+#if defined(__clang__) && defined(__i386__)
+#define	__atomic_load_n(ptr, order) __sync_fetch_and_add(ptr, 0)
+#define	__atomic_store_n(ptr, val, order) __sync_swap(ptr, val)
+#endif
             uint64_t* pTimestampShm = (uint64_t*)m_sseuShm + sliceCountShm;
             uint64_t   timestampShm = __atomic_load_n(pTimestampShm, __ATOMIC_SEQ_CST);
             if (sliceNum == sliceCountShm)
